ownCloud/Client/Sharing/PublicLinkEditTableViewController.swift

```diff
@@ -0,0 +1,590 @@
+			guard let expireSection = expireSection else { return }
+
+			if expireSection.row(withIdentifier: "date-picker-row") == nil {
+
+				let datePickerRow = StaticTableViewRow(datePickerWithAction: { [weak self] (row, sender) in
+
+					guard let datePicker = sender as? UIDatePicker else { return }
+					if let expireDateRow = expireSection.row(withIdentifier: "expire-date-row") {
+						OnMainThread {
+							expireDateRow.cell?.textLabel?.text = dateFormatter.string(from: datePicker.date)
+							expireDateRow.representedObject = datePicker.date
+						}
+					}
+
+					if let self = self, let core = self.core, !self.createLink {
+						guard let share = self.share else { return }
+						core.update(share, afterPerformingChanges: { (share) in
+							share.expirationDate = datePicker.date
+						}, completionHandler: { [weak self] (error, share) in
+							guard let self = self else { return }
+							if error == nil {
+								guard let changedShare = share else { return }
+								self.share?.expirationDate = changedShare.expirationDate
+
+								if let expireDateRow = expireSection.row(withIdentifier: "expire-date-row") {
+									OnMainThread {
+										expireDateRow.cell?.textLabel?.text = dateFormatter.string(from: datePicker.date)
+										expireDateRow.representedObject = datePicker.date
+									}
+								}
+							} else {
+								if let shareError = error {
+									OnMainThread {
+										let alertController = UIAlertController(with: "Setting expiration date failed".localized, message: shareError.localizedDescription, okLabel: "OK".localized, action: nil)
+										self.present(alertController, animated: true)
+									}
+								}
+							}
+						})
+					}
+				}, date: expireDate, identifier: "date-picker-row")
+				expireSection.add(row: datePickerRow, animated: true)
+			} else {
+				if let datePickerRow = expireSection.row(withIdentifier: "date-picker-row") {
+					expireSection.remove(rows: [datePickerRow], animated: true)
+				}
+			}
+		}, title: dateFormatter.string(from: expireDate), style: .plain, alignment: .left, identifier: "expire-date-row")
+
+		expireSection.add(row: expireDateRow)
+	}
+
+	// MARK: - Action Section
+
+	func addActionSection() {
+		let dateFormatter = DateFormatter()
+		dateFormatter.dateStyle = .medium
+		dateFormatter.timeStyle = .short
+		var footer = ""
+		if let date = share?.creationDate {
+			footer = String(format: "Shared since: %@".localized, dateFormatter.string(from: date))
+		}
+
+		let deleteSection = StaticTableViewSection(headerTitle: nil, footerTitle: footer)
+
+		if let share = self.share, let shareURL = share.url {
+			deleteSection.add(rows: [
+				StaticTableViewRow(headerRowWithAction: { (_, _) in
+					UIPasteboard.general.url = shareURL
+				}, headerTitle: String(format:"%@", shareURL.absoluteString), title: "Copy Public Link".localized, accessoryView: .none)
```

Comment:
Would IMO look better and make it less busy to
- have that section with a header "Public Link"
- put the link url into a regular row
- have a "Copy" button on the right side of that row.
- have the delete row just labelled "Delete" then